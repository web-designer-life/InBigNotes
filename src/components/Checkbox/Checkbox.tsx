import React from 'react';
import { 
    CheckboxContainer, 
    Icon, 
    HiddenCheckbox, 
    StyledCheckbox, 
} from './style';
import checkMark from '../../images/check-mark.svg';

interface Props {
    className?: string,
    checked: boolean,
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void,
    labelWrap?: boolean,
};

const Checkbox: React.FC<Props> = ({
    checked,
    labelWrap = true,
    ...props
}) => {
    const content = (
        <CheckboxContainer>
            <HiddenCheckbox checked={checked} {...props} />
            <StyledCheckbox checked={checked}>
                <Icon src={checkMark}/>
            </StyledCheckbox>
        </CheckboxContainer>
    );

    return <label>{content}</label>;
};

export default Checkbox;
