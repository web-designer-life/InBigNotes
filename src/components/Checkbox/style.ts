import styled from 'styled-components';

const CheckboxContainer = styled.div`
    display: inline-block;
    vertical-align: middle;
`;

const Icon = styled.svg`
    width: 54px;
    height: 40px;
    fill: none;
`;

const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
    border: 0;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    white-space: nowrap;
    width: 1px;
`;

const StyledCheckbox = styled.div<{ checked: boolean }>`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 80px;
    height: 80px;
    background: #FFFFFF;
    border-radius: 20px;
    border: 5px solid #0047FF;
    transition: all 150ms;

    ${Icon} {
        visibility: ${(props) => (props.checked ? "visible" : "hidden")};
    }
`;

export { CheckboxContainer, Icon, HiddenCheckbox, StyledCheckbox };