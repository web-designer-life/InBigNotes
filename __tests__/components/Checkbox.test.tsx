import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Checkbox from '../../src/components/Checkbox';

const activeCheckboxProps = {
    checked: true,
    onChange: () => { },
};

const inactiveCheckboxProps = {
    checked: false,
    onChange: () => { },
};

describe('Checkbox component', () => {
    it('should render active Checkbox component', () => {
        const component = renderer.create(<Checkbox {...activeCheckboxProps} />);
        expect(component).toMatchSnapshot();
    });

    it('should render inactive Checkbox component', () => {
        const component = renderer.create(<Checkbox {...inactiveCheckboxProps} />);
        expect(component).toMatchSnapshot();
    });
});
