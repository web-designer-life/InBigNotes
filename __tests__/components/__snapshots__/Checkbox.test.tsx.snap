// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render active Checkbox component 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c0:hover {
  cursor: pointer;
}

.c4 {
  width: 32px;
  height: 24px;
  fill: none;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 55px;
  height: 55px;
  background: #FFFFFF;
  border-radius: 15px;
  border: 4px solid #0047FF;
}

.c2 .c3 {
  visibility: visible;
}

<label
  className="c0"
>
  <input
    checked={true}
    className="c1"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    checked={true}
    className="c2"
  >
    <img
      className="c3 c4"
      src="test-file-stub"
    />
  </div>
</label>
`;

exports[`Checkbox component should render inactive Checkbox component 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c0:hover {
  cursor: pointer;
}

.c4 {
  width: 32px;
  height: 24px;
  fill: none;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c5 .c3 {
  visibility: visible;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 55px;
  height: 55px;
  background: #FFFFFF;
  border-radius: 15px;
  border: 4px solid #0047FF;
}

.c2 .c3 {
  visibility: hidden;
}

<label
  className="c0"
>
  <input
    checked={false}
    className="c1"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    checked={false}
    className="c2"
  >
    <img
      className="c3 c4"
      src="test-file-stub"
    />
  </div>
</label>
`;
